src/qt/transactionrecord.cpp

```diff
@@ -223,11 +225,6 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet*
             sub.type = TransactionRecord::SendToSelf;
             sub.address = "";
 
-            for (unsigned int nOut = 0; nOut < wtx.vout.size(); nOut++) {
-                const CTxOut& txout = wtx.vout[nOut];
-                sub.idx = parts.size();
```

Comment:
where are you setting sub.idx for this record now that this was removed?

Reply:
there isn't any (single) one here, the unique record comprehends all tx outputs and inputs in this case. 
So it remains at the default value `0`.